compiler/luci/pass/src/FuseSliceWithTConvPass.cpp

```diff
@@ -0,0 +1,313 @@
+  if (!filter)
+    return luci::Padding::UNDEFINED;
+
+  auto tconv_shape = dynamic_cast<luci::CircleConst *>(tconv->inputSizes());
+  if (!tconv_shape)
+    return luci::Padding::UNDEFINED;
+
+  luci::Padding padding = luci::Padding::UNDEFINED;
+  std::initializer_list<luci::Padding> paddings_to_check = {luci::Padding::VALID,
+                                                            luci::Padding::SAME};
+
+  auto const filter_height = filter->dim(1).value();
+  auto const filter_width = filter->dim(2).value();
+  auto const stride_height = tconv->stride()->h();
+  auto const stride_width = tconv->stride()->w();
+
+  for (auto padding_to_check : paddings_to_check)
+  {
+    auto const in_height =
+      compute_input_size(padding_to_check, out_height, filter_height, stride_height);
+    auto const pad_top_virtual =
+      extract_pad_value(stride_height, in_height, filter_height, out_height);
+    if (pad_top_virtual != pad_top)
+      continue;
+
+    auto const in_width =
+      compute_input_size(padding_to_check, out_width, filter_width, stride_width);
+    auto const pad_left_virtual =
+      extract_pad_value(stride_width, in_width, filter_width, out_width);
+    if (pad_left_virtual == pad_left)
+    {
+      padding = padding_to_check; // adissible paddding is found
+      break;
+    }
+  }
+
+  return padding;
+}
+
+/**
+ *  Fuse Slice with CircleTransposeConv if possible
+ *
+ *  NOTE: In case predecessor of slice is tconv, we can try to merge slice with tconv,
+ *  because spatial slice is reduction so as padding for tconv,
+ *  while channels slice reduction can be directly modeled in tconv.
+ *  For now there is no option to set explicitely pad values for
+ *  CircleTransposeConv. Only using VALID/SAME and output shape is the only way
+ *  to set pad values. That is why not all numerical values of pad are legal for such
+ *  transform.
+ *
+ *  BEFORE
+ *                    |
+ *           [CircleTransposeConv]
+ *                    |
+ *               [CircleSlice]
+ *                    |
+ *
+ *  AFTER (if possible)
+ *                    |
+ *            [CircleTransposeConv] (with m.b. changed padding, output shape, and filter/bias)
+ *                    |
+ *
+ */
+
+bool fuse_slice_with_tconv(luci::CircleSlice *slice)
+{
+  // NOTE: assume NHWC layout
+  auto tconv = dynamic_cast<luci::CircleTransposeConv *>(slice->input());
+  if (not tconv)
+    return false;
+
+  auto const tconv_users = loco::succs(tconv);
+  if (tconv_users.size() != 1)
+  {
+    // tconv is an operand by many operations (in general no fusion is possible)
+    return false;
+  }
+
+  // offset
+  auto begin = dynamic_cast<luci::CircleConst *>(slice->begin());
+  if (not begin)
+    return false;
+
+  // output shape
+  auto size = dynamic_cast<luci::CircleConst *>(slice->size());
+  if (not size)
+    return false;
+
+  if (begin->dtype() != loco::DataType::S32 && size->dtype() != loco::DataType::S32)
```

Comment:
Maybe?
```suggestion
  if (begin->dtype() != loco::DataType::S32 or size->dtype() != loco::DataType::S32)
```

Or you can write condition separately for readability.

Reply:
> Or you can write condition separately for readability.

Separately looks good.

Reply:
@jinevening 
Thank you, it was a bug, i'll use RETURN_FALSE_UNLESS for better clarity.